classes/Provider/ResetEmailerServiceProvider.php

```diff
@@ -24,12 +26,23 @@
      */
     public function register(Container $app)
     {
-        $app['reset_emailer'] = function ($app) {
+        $options = [
+            'host'     => $app->config('mail.host'),
+            'port'     => $app->config('mail.port'),
+            'username' => $app->config('mail.username'),
+            'password' => $app->config('mail.password'),
+        ];
+        $transport = (new Swift_SmtpTransport($options['host'], $options['port']))
+            ->setUsername($options['username'])
+            ->setPassword($options['password']);
+        $swiftMailer          = new Swift_Mailer($transport);
+        $app['reset_emailer'] = function ($app) use ($options, $swiftMailer) {
             return new ResetEmailer(
-                $app['mailer'],
+                $swiftMailer,
                 $app['twig'],
                 $app->config('application.email'),
-                $app->config('application.title')
+                $app->config('application.title'),
+                $options
```

Comment:
As far as I can tell, the constructor of `ResetEmailer` doesn't need this argument, can we remove it?